packages/valory/protocols/tendermint/tests/test_tendermint.py

```diff
@@ -0,0 +1,173 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2022 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Tests package for the 'valory/tendermint' protocol."""
+
+from typing import Type
+
+import pytest
+from aea.common import Address
+from aea.mail.base import Envelope
+from aea.protocols.base import Message
+from aea.protocols.dialogue.base import Dialogue as BaseDialogue
+from aea.protocols.dialogue.base import DialogueLabel
+
+from packages.valory.protocols.tendermint import TendermintMessage
+from packages.valory.protocols.tendermint.custom_types import ErrorCode
+from packages.valory.protocols.tendermint.dialogues import (
+    TendermintDialogue,
+    TendermintDialogues,
+)
+
+
+@pytest.mark.parametrize(
+    "msg",
+    [
+        TendermintMessage(
+            performative=TendermintMessage.Performative.REQUEST,  # type: ignore
+        ),
+        TendermintMessage(
+            performative=TendermintMessage.Performative.RESPONSE,  # type: ignore
+            info="",
+        ),
+        TendermintMessage(
+            performative=TendermintMessage.Performative.ERROR,  # type: ignore
+            error_code=ErrorCode.INVALID_REQUEST,
+            error_msg="",
+            error_data={},
+        ),
+    ],
+)
+def test_serialization(msg: TendermintMessage) -> None:
+    """Test the serialization of Tendermint speech-act messages."""
+
+    msg.to = "receiver"
+    envelope = Envelope(
+        to=msg.to,
+        sender="sender",
+        message=msg,
+    )
+    envelope_bytes = envelope.encode()
+
+    actual_envelope = Envelope.decode(envelope_bytes)
+    expected_envelope = envelope
+    assert expected_envelope.to == actual_envelope.to
+    assert expected_envelope.sender == actual_envelope.sender
+    assert (
+        expected_envelope.protocol_specification_id
+        == actual_envelope.protocol_specification_id
+    )
+    assert expected_envelope.message != actual_envelope.message
```

Comment:
Is there an issue for this?

```suggestion
    assert expected_envelope.message == actual_envelope.message
```

Reply:
https://github.com/valory-xyz/open-aea/issues/495